test/stringify.js

```diff
@@ -261,4 +262,45 @@ test('stringify()', function (t) {
         st.equal(qs.stringify({ a: 'a', z: { zj: {zjb: 'zjb', zja: 'zja'}, zi: {zib: 'zib', zia: 'zia'} }, b: 'b' }, { sort: null, encode: false }), 'a=a&z[zj][zjb]=zjb&z[zj][zja]=zja&z[zi][zib]=zib&z[zi][zia]=zia&b=b');
         st.end();
     });
+
+    t.test('can stringify with custom encoding', function (st) {
+        st.equal(qs.stringify({ 県: '大阪府', '': ''}, {
+            encoder: function (str) {
+                if (str.length === 0) {
+                    return '';
+                }
+                var buf = iconv.encode(str, 'shiftjis');
+                var result = [];
+                for (var i=0; i < buf.length; ++i) {
+                    result.push(buf.readUInt8(i).toString(16));
+                }
+                return '%' + result.join('%');
+            }
+        }), '%8c%a7=%91%e5%8d%e3%95%7b&=');
+        st.end();
+    });
+
+    t.test('throws error with wrong encoder', function (st) {
+        try {
+            qs.stringify({}, {
+                encoder: 'string'
+            });
+        } catch (e) {
+            st.equal(e.message, 'Encoder has to be a function.');
+            st.end();
+            return;
+        }
+    });
+
+    t.test('can use custom encoder for a buffer object', function (st) {
```

Comment:
let's make sure to skip this test if `Buffer` isn't available, since these tests need to pass in browsers and old node versions too. eg `t.test('blah', { skip: condition }, function (st) {`